packages/2d/src/Components/GIF.ts

```diff
@@ -0,0 +1,152 @@
+import { AnimationAPI, AnimationFrame } from './Animation';
+import { useType } from '@hex-engine/core';
+import gifken, { Gif } from 'gifken';
+
+interface GIFInterface extends AnimationAPI<HTMLImageElement> {
+  getGif(): Gif,
+  drawCurrentFrame(context: CanvasRenderingContext2D, x?: number, y?: number): void;
+}
+
+/**
+ * A component which able you to play and manipulate gifs into hex-engine.
+ * @example
+ * import someGifFile from "./your.gif";
+ *
+ * export default function MyGif() {
+ *   useType(MyGif);
+ *
+ *   const gif = useNewComponent(() => GIF({
+ *     url: someGifFile, 
+ *     width: 200, 
+ *     height: 200, 
+ *     fps: 20,
```

Comment:
It is possible for GIFs to have frames with different lengths; one frame can be long and the other can be short. Does gifken support this?

If you change the code to use the [Animation](https://github.com/suchipi/hex-engine/blob/master/packages/2d/src/Components/Animation.ts) component, you can specify the duration of each frame, and it will handle advancing to the next frame at the right time for you.

Reply:
I have find in the gifken documentation a property called [delayCentiSeconds](https://aaharu.github.io/gifken/classes/_gifframe_.gifframe.html#delaycentiseconds.)
I don't know if is exaclty what we want but i can test to set this property as duration when i will update the GIF component with the Animation component.

Reply:
Alright, sounds good! You can use https://ezgif.com/maker to make a GIF that has frames with differing durations.